src/models/model.mc

```diff
@@ -0,0 +1,71 @@
+include "types/btree.mc"
+include "types/dfa.mc"
+include "types/nfa.mc"
+
+-- Represents models that can be visualized and its associated data.
+-- Also provides toDot functions for all types defined below
+type Model
+    con Digraph : (Digraph, vertex2str, edge2str,    displayNames) -> Model
+    con Digraph : (Digraph, vertex2str, edge2str                 ) -> Model
+    con DFA     : (DFA, input, state2str, label2str, displayNames) -> Model
+    con DFA     : (DFA, input, state2str, label2str              ) -> Model
+    con Graph : (Graph,  vertex2str, edge2str,       displayNames) -> Model
+    con Graph : (Graph,  vertex2str, edge2str                    ) -> Model
+    con NFA     : (NFA,input, state2str,  label2str, displayNames) -> Model
+    con NFA     : (NFA,input, state2str,  label2str              ) -> Model
+    con BTree   : (BTree,node2str,                   displayNames) -> Model
+    con BTree   : (BTree,node2str                                ) -> Model
```

Comment:
Having several constructors of the same name is rather funky and won't work with a proper type system. Wouldn't it be better to just always include `displayNames`, but have it be an empty list if the user doesn't want to add any names? That way you don't have to duplicate match cases in the visualizing function either.

Reply:
This has been updated and only the constructors that use displayNames are left. Good point!